### PR TITLE
raise NotImplementedError instead of return NotImplemented

### PR DESCRIPTION
**Pull request by r-barnes**

### CODE REVIEW

**Comment by r-barnes** [comment]
Test failures appear to be flakes?

File: cvxpy/interface/scipy_wrapper.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def wrap_bin_op(method):
     """
     def new_method(self, other):
         if isinstance(other, exp.Expression):
-            return NotImplemented
```
Comment (SteveDiamond) [inline cvxpy/interface/scipy_wrapper.py:29]: This one you can't change. It helps with operator overloading.

**Comment by SteveDiamond** [comment]
Yeah sorry about the confusing tests. We're in the process of revamping our CI. The AppVeyor test is real. See my comment.

File: cvxpy/interface/scipy_wrapper.py (hunk 1 above)
Comment (r-barnes) [reply cvxpy/interface/scipy_wrapper.py]: Thanks; reverted.

**Comment by SteveDiamond** [comment]
LGTM. Thanks for doing this!

**Pull request merged**